### PR TITLE
Chore: replace React.FC<Props> with simple function component

### PR DESCRIPTION
**Pull request by ryantxu**
See also https://github.com/facebook/create-react-app/pull/8177

Internally, we have discussed that we should prefer simple react function components over the explicitly typed `React.FC<Props>`, see the link above for reasons.

I am a creation of habit, and tend to just copy paste from similar code... but lots of our code uses this pattern.   This PR:
1. Updates the style guides to suggest the new structure
2. replaces the syntax in most instances
3. does not touch anythign where the Props include `children`

### CODE REVIEW

File: docs/sources/developers/plugins/migration-guide.md [hunk 1]
```diff
@@ -708,9 +708,9 @@ import { PanelProps } from '@grafana/data';
 
 interface Props extends PanelProps<SimpleOptions> {}
 
-export const MyPanel: React.FC<Props> = ({ options, data, width, height }) => {
+export function MyPanel({ options, data, width, height }: Props) {
   // ...
-};
+}
```
Comment (itsmylife) [inline docs/sources/developers/plugins/migration-guide.md:713]: is there a particular reason why you choose regular function definition over arrow function?

**Review by itsmylife — APPROVED** [review]
LGTM

File: contribute/style-guides/styling.md [hunk 2]
```diff
@@ -113,5 +113,5 @@ const ComponentA: React.FC<Props> = ({ className }) => {
   );
 
   return <div className={finalClassName}>As red as you can ge</div>;
-};
+}
```
Comment (nmarrs) [inline contribute/style-guides/styling.md:116]: For style guide maybe we can include two example components, first with `function Component() { }` declaration and the second with `const Component = () => { }` declaration given we have both forms throughout codebase? (i.e. both are acceptable ways of defining a component)

**Review by nmarrs — APPROVED** [review]
image

**Pull request merged**